### PR TITLE
flavors: Raise conflict if CPU or Memory resource requests are present when using flavors

### PR DESCRIPTION
**Pull request by lyarwood**
**What this PR does / why we need it**:

Previously a user could request both the use of a flavor that defines a guest visible amount of CPU/Memory *and* separate resource requests for CPU and memory within the associated `VirtualMachine`. 

This obviously isn't desirable as  there is no guarantee these requests could work together. For now a conflict should be raised. In the future flavors should be extended to support both methods of defining resources, allowing us to ensure that the requests are viable before application to the `VirtualMachineInstance`.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Conflicts will now be raised when using flavors if the `VirtualMachine` defines any `CPU` or `Memory` resource requests.
```

### CODE REVIEW

**Review by stu-gott — COMMENTED** [review]
Overall this looks good.

File: tests/flavor_test.go [hunk 1]
```diff
@@ -292,6 +295,55 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
 			Expect(cause.Field).To(Equal("spec.template.spec.domain.cpu"))
 		})
 
+		DescribeTable("[test_id:TODO] should fail if the VirtualMachine has ", func(resources virtv1.ResourceRequirements, expectedField string) {
+			vmi := tests.NewRandomVMI()
```
[the remark this answers is not in the document]
Comment (lyarwood) [reply tests/flavor_test.go:299]: ACK done, thanks!

**Review by davidvossel — COMMENTED** [review]
/approve
/lgtm

**Pull request merged**